microsoft/botbuilder-dotnet
# Fix Random thread safety and control random seed to improve testability

Fixes #4361

## Description
This PR aims to accept a "Microsoft.SetTestOptions" event, and apply the options to SDK.

## Specific Changes
1. Add `setTestOptions` event
2. Save TestOptions into the conversation state. Including the Random seed and random value.
3. Achieve the state from "dialogContext.State", and apply it to SDK code.
4. Adjust random generator code to use the random in "TestOptions".

## How to test
1. Start project "Microsoft.Bot.Builder.TestBot.Json"
2. Input 26 to see the dialog output.

## Progress
- [x] Adaptive-expression
- [x] LG Evaluator/Expander
- [ ] Random selector
